# Student Marks Dataset Analysis

## Week 1 – Python for Data Science Basics

### Objective

The objective of this project is to analyze a Student Marks dataset using Python, Pandas, and NumPy. The project focuses on basic data exploration, checking and handling missing data, and calculating the average, maximum, and minimum scores for Mathematics, Reading, and Writing.


Dataset Description

The dataset contains performance information for 1,000 students across different demographic and educational categories.

The dataset contains 8 columns:

Gender
Race/Ethnicity
Parental Level of Education
Lunch
Test Preparation Course
Math Score
Reading Score
Writing Score

The three score columns are numerical and are used for the main analysis.

In [57]:
import pandas as pd
import numpy as np

In [58]:
df = pd.read_csv("/StudentsPerformance.csv")

In [59]:
df.head()

,gender,race/ethnicity,parental level of education,lunch,test preparation course,math score,reading score,writing score
0,female,group B,bachelor's degree,standard,none,72,72,74
1,female,group C,some college,standard,completed,69,90,88
2,female,group B,master's degree,standard,none,90,95,93
3,male,group A,associate's degree,free/reduced,none,47,57,44
4,male,group C,some college,standard,none,76,78,75


## Data Exploration

The dataset was inspected to understand its structure, number of records, column names, and data types.


In [60]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


In [61]:
df.isnull().sum()

,0
gender,0
race/ethnicity,0
parental level of education,0
lunch,0
test preparation course,0
math score,0
reading score,0
writing score,0


### Missing Data Check

The dataset was checked for missing values using `isnull().sum()`. All columns contain 1,000 non-null values, indicating that there are no missing values in the dataset. Therefore, no rows needed to be removed.


## Score Analysis

The average, maximum, and minimum scores were calculated for Mathematics, Reading, and Writing using Pandas.


In [62]:
results = pd.DataFrame({
    "Average": df[["math score", "reading score", "writing score"]].mean(),
    "Maximum": df[["math score", "reading score", "writing score"]].max(),
    "Minimum": df[["math score", "reading score", "writing score"]].min()
})
results

,Average,Maximum,Minimum
math score,66.089,100,0
reading score,69.169,100,17
writing score,68.054,100,10


## NumPy Analysis

NumPy was used to calculate the overall average score across the three subjects. The three score columns were converted into a NumPy array for numerical analysis.


In [63]:
scores = df[["math score", "reading score", "writing score"]].to_numpy()

overall_average = np.mean(scores)

overall_average

np.float64(67.77066666666667)

## Findings and Insights

The analysis shows that the Reading score had the highest average at 69.169, followed by Writing at 68.054 and Mathematics at 66.089. The maximum score in all three subjects was 100.

The minimum scores were 0 in Mathematics, 17 in Reading, and 10 in Writing. The overall average across the three subjects was approximately 67.77.

The dataset contained no missing values, so no rows needed to be removed during the data-cleaning process.


## Conclusion

This project demonstrated the basic use of Python, Pandas, and NumPy for data analysis. The Student Marks dataset was successfully loaded, explored, checked for missing values, and analyzed. Pandas was used to calculate and display the average, maximum, and minimum scores, while NumPy was used for numerical analysis. The project provided practical experience in basic data cleaning and extracting useful insights from a dataset.
